modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/Fields.tsx

```diff
@@ -0,0 +1,493 @@
+import ClayModal from '@clayui/modal';
+import {ManagementToolbar} from 'frontend-js-components-web';
+import {fetch, navigate, openModal, openToast} from 'frontend-js-web';
+import React, {useEffect, useRef, useState} from 'react';
+
+import {API_URL, OBJECT_RELATIONSHIP} from '../Constants';
+import {FDSViewSectionInterface} from '../FDSView';
+import {FDSViewType} from '../FDSViews';
+import OrderableTable from '../components/OrderableTable';
+
+type FDSFieldType = {
+	label: string;
+	name: string;
+	renderer: string;
+	sortable: boolean;
+	type: string;
+};
+
+type FieldType = {
+	name: string;
+	selected: boolean;
+	type: string;
+	visible: boolean;
+};
+
+interface AddFDSFieldsModalContentInterface {
+	closeModal: Function;
+	fdsView: FDSViewType;
+	onSave: Function;
+}
+
+const AddFDSFieldsModalContent = ({
+	closeModal,
+	fdsView,
+	onSave,
+}: AddFDSFieldsModalContentInterface) => {
+	const [fields, setFields] = useState<Array<FieldType> | null>(null);
+	const [query, setQuery] = useState('');
+
+	const onSearch = (query: string) => {
+		setQuery(query);
+
+		if (!fields) {
+			return;
+		}
+
+		const regexp = new RegExp(query, 'i');
+
+		setFields(
+			fields.map((field) => ({
+				...field,
+				visible: Boolean(field.name.match(regexp)),
+			}))
+		);
+	};
+
+	useEffect(() => {
+		const getFields = async () => {
+			const {restApplication, restSchema} = fdsView[
+				OBJECT_RELATIONSHIP.FDS_ENTRY_FDS_VIEW
+			];
+
+			const response = await fetch(`/o${restApplication}/openapi.json`);
+
+			if (!response.ok) {
+				openToast({
+					message: Liferay.Language.get(
+						'your-request-failed-to-complete'
+					),
+					type: 'danger',
+				});
+
+				return null;
+			}
+
+			const responseJSON = await response.json();
+
+			const properties =
+				responseJSON?.components?.schemas[restSchema]?.properties;
+
+			if (!properties) {
+				openToast({
+					message: Liferay.Language.get(
+						'your-request-failed-to-complete'
+					),
+					type: 'danger',
+				});
+
+				return null;
+			}
+
+			const fieldsArray: Array<FieldType> = [];
+
+			const isObjectSchema =
+				responseJSON.components.schemas[restSchema].xml.name ===
+				'ObjectEntry';
+
+			Object.keys(properties).forEach((propertyKey) => {
+				const propertyValue = properties[propertyKey];
+
+				if (isObjectSchema && !propertyValue.extensions) {
+					return;
+				}
+
+				if (propertyKey === 'x-class-name') {
+					return;
+				}
+
+				const type = propertyValue.type;
+
+				if (type === 'object' || type === 'array') {
+					return;
+				}
+
+				fieldsArray.push({
+					name: propertyKey,
+					selected: false,
+					type,
+					visible: true,
+				});
+			});
+
+			setFields(fieldsArray);
+		};
+
+		getFields();
+	}, [fdsView]);
+
+	const isSelectAllChecked = () => {
+		if (!fields) {
+			return false;
+		}
+
+		const selectedFieldsCount =
+			fields.filter((field) => field.selected)?.length || 0;
+
+		return selectedFieldsCount === fields.length;
```

Comment:
`.filter()` returns a new array, so you are comparing `Array[] === integer` which will always be false.

Am I missing something here?

Reply:
It's chaining that might be confusing you @bryceosterhaus 

I don't really like chaining in the way @markocikos did it here but I believe this is what he was going for

```js
const selectedFields = fields.filter((field) => field.selected);

const selectedFieldsCount = selectedFields?.length || 0;

return selectedFieldsCount === fields.length;
```

Reply:
Ah I see, I didn't notice the `)?.`

Maybe we could separate it on multiple lines like your example then, I think it would help readability